x/evm/keeper/state_transition_test.go

```diff
@@ -395,76 +440,52 @@ func (suite *KeeperTestSuite) TestRefundGas() {
 			noError:        true,
 			expGasRefund:   params.TxGas / params.RefundQuotient,
 		},
-		{
-			name:           "invalid Gas value in msg",
```

Comment:
is this case not valid anymore? 🤔

Reply:
Good catch! I had deleted because thought that gas did not affected the `RefundGas` function but after a second look I realize this was actually changing GasPrice and not GasLimit so it was relevant. Updated the code and added the tests back!